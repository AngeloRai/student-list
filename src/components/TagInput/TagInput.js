const TagInput = ({ handleSubmit, setTagName, tagName }) => {
  return (
    <div>
      <form onSubmit={handleSubmit}>
        <input
          placeholder="Add a tag"
          className="ml-3 text-sm appearance-none sticky focus:outline-none focus:border-b-[2px] focus:border-gray-700 border-b-[2px] p-2"
          type="text"
          value={tagName}
          onChange={(e) => setTagName(e.target.value)}
        />
        <button type="submit"/>
      </form>
    </div>
  );
};

export default TagInput;
