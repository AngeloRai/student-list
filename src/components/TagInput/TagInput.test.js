import React from "react";
import { shallow, mount } from "enzyme";
import TagInput from "./TagInput";
import toJson from "enzyme-to-json";

describe("tests tag input component", () => {
  let wrapper;
  const setState = jest.fn();
  const mockPlaceholder = "Add a tag";
  const useStateSpy = jest.spyOn(React, "useState");
  useStateSpy.mockImplementation((init) => [init, setState]);

  beforeEach(() => {
    wrapper = mount(
      shallow(
        <TagInput placeholder={mockPlaceholder} setTagName={setState} />
      ).get(0)
    );
  });

  it("renders without crashing", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("accepts props", () => {
    const wrapper = mount(
      <TagInput placeholder={mockPlaceholder} setTagName={setState} />
    );
    expect(wrapper.props().placeholder).toBe(mockPlaceholder);
    expect(wrapper.props().setTagName).toBe(setState);
  });

  it("Should capture tag input correctly onChange", () => {
    const tagNmae = wrapper.find("input").at(0);
    tagNmae.instance().value = "Test";
    tagNmae.simulate("change");
    expect(setState).toHaveBeenCalledWith("Test");
  });
});
