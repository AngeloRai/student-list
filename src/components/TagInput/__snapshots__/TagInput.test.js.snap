// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests tag input component renders without crashing 1`] = `
<div>
  <form>
    <input
      className="ml-3 text-sm appearance-none sticky focus:outline-none focus:border-b-[2px] focus:border-gray-700 border-b-[2px] p-2"
      onChange={[Function]}
      placeholder="Add a tag"
      type="text"
    />
    <button
      type="submit"
    />
  </form>
</div>
`;
